File: common-rule-engine/src/test/files/checks/OctalNumberCheck/OctalNumberCheck.go
package Foo

var a = 0
var b = 1
var c = 42
var mode = 0644 // 4-digit octal numbers are ignored as are often used for file permissions
var d = 010 // Noncompliant {{Use decimal rather than octal values.}}
//      ^^^
var e = 00 // Noncompliant
var f = 0.
var g = 0x00
var h = 0X00

func foo() {
  foo(010)   // Noncompliant
}

func ignoreBitwiseOperations() {
  var a = 0
  a &= 02
  a |= 02
  a = a & 02
  a = a &^ 02
  a = a | 02
  a = a << 02
  a = a >> 02
}

File: common-rule-engine/src/test/files/checks/OctalNumberCheck/OctalNumberCheck.java
class A {
  int a = 0;
  int b = 1;
  int c = 42;
  int mode = 0644; // 4-digit octal numbers are ignored as are often used for file permissions
  int d = 010; // Noncompliant {{Use decimal rather than octal values.}}
//        ^^^
  int e = 00; // Noncompliant
  int f = 0.;
  int g = 0x00;
  int h = 0X00;
  int j = 0b0101;
  int k = 0B0101;

  void foo() {
    int x = 010; // Noncompliant
  }
}
